Firestore/core/src/model/value_util.cc

```diff
@@ -736,6 +753,62 @@ bool IsMaxValue(const google_firestore_v1_Value& value) {
              kRawMaxValueFieldValue;
 }
 
+int64_t IndexOfKey(const google_firestore_v1_MapValue& mapValue,
```

Comment:
Can we use optional<int64_t> instead of special value -1?

Reply:
Yes, implemented. This is a nice improvement.